### DISCUSSION TITLE
Poco 1.13.1 POCO_DATA_NO_SQL_PARSER

### DISCUSSION THREAD

**Discussion by micheleselea** [pocoproject/poco#4452] [Q&A]
In default Config.h we have
```
// Uncomment to explicitly disable SQLParser
// #define POCO_DATA_NO_SQL_PARSER
```
But ActiveRecord do not compile because it does not found
`#include "sql-parser/src/SQLParser.h"`
I'm going to explicit define POCO_DATA_NO_SQL_PARSER, but what is "sql-parser/src/SQLParser.h" is a system lib?

**Comment by aleks-f** [comment]
The platform, I assume, is windows with VS 2019. VS2022 builds fine. 

Use cmake to generate VS projects, or fix 2019 projects using [progen](https://github.com/pocoproject/poco/wiki/Generating-VS-Projects-using-ProGen) and send a PR

> what is "sql-parser/src/SQLParser.h" is a system lib?

I don't understand the question.

**Comment by micheleselea** [comment]
I'm using VS2022 and when I compile Data the POCO_DATA_NO_SQL_PARSER define is NOT defined by default, so the
`#include "Poco/Data/SQLParser.h"`
includes
`#include "sql-parser/src/SQLParser.h"`
I do not have that file anywere so I have to compile with #define POCO_DATA_NO_SQL_PARSER 
is that the correct behavior?

**Comment by aleks-f** [comment]
What do you mean you don't have it anywhere? It's right here: https://github.com/pocoproject/poco/tree/poco-1.13.1/Data/src/sql-parser

To get help, you need to give sufficient information. Where do your sources come from?

**Comment by micheleselea** [comment]
Ok so I think something wrong with the ZIP file. I get the zip from this link
https://pocoproject.org/releases/poco-1.13.1/poco-1.13.1-all.zip
and I did not found the sql-parser directory

**Comment by micheleselea** [comment]
![image](https://github.com/pocoproject/poco/assets/5929871/318a069d-0903-4776-a302-1ed849960b92)

**Comment by aleks-f** [comment]
https://github.com/pocoproject/poco/issues/4453

**Comment by micheleselea** [comment]
Anyway I did not yet figure out what the POCO_DATA_NO_SQL_PARSER change from (for example) Poco::Data::SQLite point of view, does it change anything? it something about performance?

**Comment by aleks-f** [comment]
It [parses](https://github.com/pocoproject/poco/blob/578a481cce2b2db3ea969caf62674369008e0dd2/Data/src/Statement.cpp#L174-L189) the sql statement. Can be [disabled](https://github.com/pocoproject/poco/blob/578a481cce2b2db3ea969caf62674369008e0dd2/Data/include/Poco/Data/AbstractSessionImpl.h#L291) at runtime; it was needed to be able to deal with transactions (not start a transaction in non-autocommit mode automatically for select-only statements)
